batchflow/monitor.py

```diff
@@ -234,7 +234,7 @@ def __init__(self, function=None, frequency=0.1, gpu_list=None, **kwargs):
         super().__init__(function=function, frequency=frequency, **kwargs)
 
         # Fallback to env variable
-        if gpu_list is None:
+        if gpu_list is None or len(gpu_list) == 0:
```

Comment:
`if not gpu_list`? seems like it is one of the cases where it is truly appropriate